packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -225,6 +229,27 @@ class SliverConstraints extends Constraints {
   /// contents depends on the [growthDirection].
   final double scrollOffset;
 
+  /// The scroll distance that has been consumed by all [Sliver]s that came
+  /// before this [Sliver].
+  ///
+  /// [Sliver]s often lazily create their internal content as layout occurs,
+  /// e.g., [SliverList]. In this case, when [Sliver]s exceed the viewport,
+  /// their children are built lazily, and the [Sliver] does not have enough
+  /// information to estimate its total extent, [consumedScrollExtent] will be
+  /// [double.infinity] for all [Sliver]s that appear after the lazily
```

Comment:
I read halfway through the second paragraph to realize that you're not really continuing to elaborate on the first paragraph, you're pointing out exceptions to the first paragraph. Add a `## Edge cases` section header or some such.

Reply:
done